corehq/ex-submodules/casexml/apps/phone/models.py

```diff
@@ -354,6 +355,53 @@ def test_only_get_dependent_cases_on_phone(self):
         raise NotImplementedError()
 
 
+def save_synclog_to_sql(synclog_json_object):
+    synclog = synclog_to_sql_object(synclog_json_object)
+    synclog.save()
+
+
+def synclog_to_sql_object(synclog_json_object):
+    # synclog_json_object should be a SyncLog instance
+    # if synclog_json_object._id
+    synclog = None
+    if synclog_json_object._id:
+        synclog = SyncLogSQL.objects.filter(synclog_id=synclog_json_object._id).first()
+
+    is_new_synclog_sql = not synclog_json_object._id or not synclog
+
+    if is_new_synclog_sql:
+        synclog_id = synclog_json_object._id or uuid.uuid4().hex.lower()
+        synclog_json_object._id = synclog_id
+        synclog = SyncLogSQL(
+            domain=synclog_json_object.domain,
+            user_id=synclog_json_object.user_id,
+            synclog_id=synclog_id,
+            date=synclog_json_object.date,
+            previous_synclog_id=getattr(synclog_json_object, 'previous_synclog_id', None),
+            log_format=synclog_json_object.log_format,
+        )
+    synclog.doc = synclog_json_object.to_json()
+    return synclog
+
+
+class SyncLogSQL(models.Model):
+    domain = models.CharField(max_length=255, null=True, blank=True, default=None, db_index=True)
+    user_id = models.CharField(max_length=255, default=None, db_index=True)
+    date = models.DateTimeField(db_index=True, null=True, blank=True)
+    synclog_id = models.UUIDField(unique=True, primary_key=True)
+    # needs to be a foreign key?
+    previous_synclog_id = models.CharField(max_length=255, default=None, null=True, blank=True)
+    doc = JSONField()
+    log_format = models.CharField(
+        max_length=10,
+        default=LOG_FORMAT_LEGACY,
+        choices=[
+            (format, format)
+            for format in [LOG_FORMAT_LEGACY, LOG_FORMAT_SIMPLIFIED, LOG_FORMAT_LIVEQUERY]
+        ]
+    )
```

Comment:
There are some other fields on SyncLogs that could be useful to keep:
* build_id
* duration
* last_submitted
* had_state_error
* error_date
* error_hash